scripts/gulp/minify-stream.js

```diff
@@ -0,0 +1,56 @@
+/* eslint-env node */
+'use strict';
+
+const { Transform } = require('stream');
+
+const terser = require('terser');
+
+/**
+ * Return a Node `Transform` stream that minifies JavaScript input.
+ *
+ * This is designed to be used both to process individual modules as a Browserify
+ * transform, and also to be applied to the output of the whole bundle to
+ * compress the module infrastructure that Browserify adds.
+ *
+ * @example
+ *   browserify(['src/app.js'])
+ *     .transform({ global: true }, minifyStream) // Minify individual modules
+ *     .pipe(minifyStream()) // Minify the code added by Browserify
+ *     .pipe(output);
+ */
+function minifyStream() {
```

Comment:
Lots of copied code, but I'm still unclear if we will ever minify, but I don't quit have the full context in my head. It almost might make sense if we start moving universal gulp tasks and subsequent functions upstream to the toolkit (for example). Maybe that could come down the road.